### PR TITLE
fix: Consider double numbers without fractional part as integers

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

**Review by jimgrace — COMMENTED** [review]
Is there a Jira ticket for this? What problem is it solving?

File: src/main/java/org/hisp/dhis/antlr/operator/AntlrComputeFunction.java [hunk 1]
```diff
@@ -62,7 +62,15 @@ public final Object evaluate( ExprContext ctx, AntlrExpressionVisitor visitor )
             values.add( value );
         }
 
-        return compute( values );
+        Object computedObject = compute(values);
```
Comment (jimgrace) [inline src/main/java/org/hisp/dhis/antlr/operator/AntlrComputeFunction.java:65]: Please add spaces and newlines per DHIS2 code formatting standards

**Comment by enricocolasante** [comment]
> Is there a Jira ticket for this? What problem is it solving?

Hey Jim,
There is no Jira for it.
The main issue for me is that if we evaluate a simple integer number, the result is a double number. This is causing some headaches in the rule engine part.
Said that, I also think is reasonable to have any expression like `1 + 1` to have a result as `2` instead of `2.0`

**Comment by enricocolasante** [comment]
@jimgrace I just added some tests

**Pull request closed without merging**